lib/kubernetes-deploy/statsd.rb

```diff
@@ -4,23 +4,41 @@
 
 module KubernetesDeploy
   class StatsD
+    extend ::StatsD
+
+    PREFIX = "KubernetesDeploy"
+
     def self.duration(start_time)
       (Time.now.utc - start_time).round(1)
     end
 
     def self.build
-      ::StatsD.default_sample_rate = 1.0
-      ::StatsD.prefix = "KubernetesDeploy"
+      self.default_sample_rate = 1.0
+      self.prefix = "KubernetesDeploy"
 
       if ENV['STATSD_DEV'].present?
-        ::StatsD.backend = ::StatsD::Instrument::Backends::LoggerBackend.new(Logger.new($stderr))
+        self.backend = ::StatsD::Instrument::Backends::LoggerBackend.new(Logger.new($stderr))
       elsif ENV['STATSD_ADDR'].present?
         statsd_impl = ENV['STATSD_IMPLEMENTATION'].present? ? ENV['STATSD_IMPLEMENTATION'] : "datadog"
-        ::StatsD.backend = ::StatsD::Instrument::Backends::UDPBackend.new(ENV['STATSD_ADDR'], statsd_impl)
+        self.backend = ::StatsD::Instrument::Backends::UDPBackend.new(ENV['STATSD_ADDR'], statsd_impl)
       else
-        ::StatsD.backend = ::StatsD::Instrument::Backends::NullBackend.new
+        self.backend = ::StatsD::Instrument::Backends::NullBackend.new
       end
-      ::StatsD.backend
+    end
+
+    def self.measure(key, value = nil, *metric_options, &block)
+      metric_options.first[:prefix] = PREFIX if metric_options.first.is_a?(Hash)
```

Comment:
I see you're copying the signature from the superclass, but I'm confused about why they're collecting the options as an array rather than a hash. It doesn't reflect what we actually expect, so it makes it much more difficult to deal with the possibilities. Can't we make the signature `(key, value=nil, **metric_options, &block)` instead? If that isn't possible for some reason, we'd need to check `value` as well because it might actually contain the options.

Reply:
Thankfully **args works here and forwards the arguments nicely on the call to super